classes/local/eventstatus_notification_helper.php

```diff
@@ -259,6 +260,48 @@ public static function notify_users_upload_queue($job, $metadata) {
         }
     }
 
+
+    /**
+     * Notify users about the failed upload being archived.
+     *
+     * @param object $job represents the upload job.
+     * @param string $title event title or filename.
+     *
+     */
+    public static function notify_users_archived_upload($job, $title) {
+        global $DB;
+        // Initialize the user list as an empty array.
+        $usertolist = [];
+        // Add uploader user object to the user list.
+        $usertolist[] = $DB->get_record('user', ['id' => $job->userid]);
+
+        // Get admin config to check if all teachers of the course should be notified as well.
+        $notifyteachers = get_config('block_opencast', 'eventstatusnotifyteachers_' . $job->ocinstanceid);
+        if ($notifyteachers) {
+            // Get the role of teachers.
+            $role = $DB->get_record('role', ['shortname' => 'editingteacher']);
+            // Get the course context.
+            $context = context_course::instance($job->courseid);
+            // Get the teachers based on their role in the course context.
+            $teachers = get_role_users($role->id, $context);
+
+            // If teachers array list is not empty, we add them to the user list.
+            if (!empty($teachers)) {
+                foreach ($teachers as $teacher) {
```

Comment:
This doesn't make sure that the uploader doesn't get notified twice, if they are also a teacher, right?

Reply:
Have you checked this line as well? or did you ask about this line?
https://github.com/Opencast-Moodle/moodle-block_opencast/blob/08d75953571034037854d0bd54e3865ce00803ab/classes/local/eventstatus_notification_helper.php#L292

Reply:
Ok, sorry, I have no idea how in the world I have overlooked that line